File: solutions/Leetcode_1704/leetcode_1704.py
class Solution1:
    def halves_are_alike(self, s: str) -> bool:
        vowels = set('aeiouAEIOU')
        half_len = len(s) // 2
        num_l, num_r = 0, 0
        for i in range(half_len):
            num_l += s[i] in vowels
            num_r += s[i+half_len] in vowels
        return num_l == num_r


if __name__ == '__main__':
    s = "AbCdEfGh"
    ans = Solution1().halves_are_alike(s)
    print(ans)

File: solutions/Leetcode_1704/leetcode_1704.cc
#include <string>
#include <iostream>

using namespace std;

class Solution1
{
public:
    bool halvesAreAlike(string s)
    {
        int half_len = s.size() / 2;
        int num_l = 0, num_r = 0;
        for (int i = 0; i < half_len; ++i)
        {
            s[i] = tolower(s[i]);
            s[i + half_len] = tolower(s[i + half_len]);
            if (s[i] == 'a' || s[i] == 'e' || s[i] == 'i' || s[i] == 'o' || s[i] == 'u')
                ++num_l;
            if (s[i + half_len] == 'a' || s[i + half_len] == 'e' || s[i + half_len] == 'i' || s[i + half_len] == 'o' || s[i + half_len] == 'u')
                ++num_r;
        }
        return num_l == num_r;
    }
};

int main()
{
    string s{"textbook"};
    bool ans = Solution1().halvesAreAlike(s);
    cout << ans << endl;
}